application/controllers/survey/index.php

```diff
@@ -118,14 +118,16 @@ public function action()
             $token = $clienttoken;
         }
 
+        if (empty($token) && isset($_SESSION['survey_'.$surveyid]['token'])) {
+            $token = $_SESSION['survey_'.$surveyid]['token'];
+        }
+
```

Comment:
What about anonymized surveys? In that case the Token information should not be available.
Also the patch should go to develop, instead.

Reply:
`{{ processString('{strip_tags(TOKEN:LASTNAME)}') }}` take care of anonymous survey. Maybe oToken are not the best candidate : 1. Can be EM accessible, 2. anonymous are issue

Reply:
anynomized surveys wont have the `$_SESSION['survey_'.$surveyid]['token']` set, or? Should not be applicable to them at all. 

Regarding the develop: im sorry, I have lost the understanding what is allowed for master and what not. I have seen a bunch of features added to master since the new-features-for-develop-only policy landed. I was not sure if it was relevant any more.

Reply:
`$_SESSION['survey_'.$surveyid]['token']` is always set to control if user is allowed to access to this survey :).

It's not related to anonymous here :)

Reply:
Even if recently someone commited a new feature it was reverted. As said, please submit for develop instead. Please make sure the token information is not available for anonymized surveys.
Thank you!

Reply:
not sure about the reverting. I have just seen regularly new features in new releases while develop branch has not been merged into master
the tail of the log:

> Changes from 3.15.3 (build 181108) to 3.15.4 (build 181109)   November 09, 2018
-New feature : list image in Survey for Brand logo file in Theme option (Denis Chenu)
>
> Changes from 3.15.1 (build 181017) to 3.15.2 (build 181107)  November 07, 2018
>
> -New feature: allow to change config file location (LouisGac)
-New feature: allow to change config file location, part 2 (LouisGac)
-New feature: configuration option to use a single upload directory for all surveys (LouisGac)
>
> Changes from 3.14.11 (build 180926) to 3.15.0 (build 181008) October 8, 2018
-New feature #14119: minute step for datetimepicker (Denis Chenu)
-New feature: Added Italian language to multilingual sample survey (Luciano Fanfoni) (Carsten Schmitz)
>
> Changes from 3.14.9 (build 180917) to 3.14.10 (build 180924) September 24, 2018
-New Feature: added Twig function getAllAnswers() (LouisGac)

Its just a bit confusing. Some new features clearly have gone straight to master and I was assuming it was ok. Can anybody explain why eg 
> Changes from 3.14.9 (build 180917) to 3.14.10 (build 180924) September 24, 2018
-New Feature: added Twig function getAllAnswers() (LouisGac)

... went straight into master & release, which in essence seems to be pretty similar to this commit?

Reply:
About 

>  -New feature : list image in Survey for Brand logo file in Theme option (Denis Chenu)

It's more because fixing related issue (user without template update rights can not upload logo in Survey theme option) . Fixing issue need this new feature :)

Reply:
We should not name it oToken - oParticipant would be better, since the Token is just an information bit of the participant.

Reply:
I did not name anything there. I'm just passing the existing things. And if I look at what It is, Its called Token_{SID} representing the class Token or TokenDynamic. 

But the class name is confusing, I agree.